### ISSUE TITLE
SemanticDB options on an sbt dotty project are wrong

### ISSUE THREAD

**Issue by adpi2** [scalameta/metals#2189]
**Describe the bug**

The sbt-metals plugin generates wrong SemanticDB options on the `Test` configuration of a Dotty project.

**To Reproduce**

1. Create an sbt dotty project
```scala
// project/plugin.sbt
addSbtPlugin("ch.epfl.lamp" % "sbt-dotty" % "0.4.4")
addSbtPlugin("org.scalameta" % "sbt-metals" % "0.9.5-SNAPSHOT")
```

```scala
// build.sbt
scalaVersion := "0.27.0-RC1"
```
2. Run `show Test / scalacOptions` in an sbt shell

```shell
sbt:example> show Test / scalacOptions
[info] * -Ysemanticdb
[info] * -semanticdb-target
[info] * /example/target/scala-0.27/meta
[info] * /example/target/scala-0.27/test-meta
```

3. Run `Test / compile` in an sbt shell

```shell
sbt:example> Test / compile
[info] compiling 4 Scala sources to /example/target/scala-0.27/test-classes ...
[error] expected file, received directory '/example/target/scala-0.27/test-meta'
[error] one error found
[error] (Test / compileIncremental) Compilation failed
[error] Total time: 0 s, completed Nov 3, 2020, 9:19:14 PM
```

**Expected behavior**

The `scalacOptions` of the Test configuration should be:

```shell
sbt:example> show Test / scalacOptions
[info] * -Ysemanticdb
[info] * -semanticdb-target
[info] * /example/target/scala-0.27/test-meta
```

**Installation:**

- Operating system: Linux
- Metals version: v0.9.5-SNAPSHOT

**Search terms**

SemanticDB, compiler options, dotty, scala 3, sbt-metals

**Comment by adpi2** [comment]
This lines of code in [sbt/sbt](https://github.com/sbt/sbt) could help to fix this issue:

https://github.com/sbt/sbt/blob/c00196abf0c0b60d301e9516b8513b498359f987/main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala#L68-L76

As you can see, the target root coming from the `Compile` configuration is removed before we add the `Test` target root.

**Comment by adpi2** [comment]
Maybe it is even easier to just do `semanticdbEnabled := true` in `sbt-metals`. It should add the required `-Ysemanticdb` and the correct `-semanticdb-target` options.

**Comment by tgodzik** [comment]
Thanks for reporting! Good catch!  I can confirm it's an issue. Should be easy to fix in the plugin.

**Comment by ckipp01** [comment]
Thanks for the report @adpi2! I should have it fixed in #2139 

> Maybe it is even easier to just do semanticdbEnabled := true in sbt-metals. It should add the required -Ysemanticdb and the correct -semanticdb-target options.

That was originally brought up in https://github.com/scalameta/metals/issues/2049, but there seemed to be concern about ensuring that the user has a supported Scala version. So we don't want to just check check the sbt version and enable it based off that which could cause issues from my understanding.
